edb/server/compiler/sertypes.py

```diff
@@ -278,7 +278,7 @@ def _describe_type(self, t, view_shapes, view_shapes_metadata,
         elif isinstance(t, s_objtypes.ObjectType):
             # This is a view
             self.schema, mt = t.material_type(self.schema)
-            base_type_id = mt.id
+            base_type_name = mt.get_displayname(self.schema)
```

Comment:
Use `get_name` instead.  `get_displayname` is intended for user-visible messages and does not identify a (possibly derived) type uniquely.

Reply:
(`get_name` gives back a `Name`, but you can just call `str` on it to get a string.)